bot/utils/__init__.py

```diff
@@ -16,50 +18,59 @@ class CaseInsensitiveDict(dict):
     """
 
     @classmethod
-    def _k(cls, key):
+    def _k(cls, key: Hashable) -> Any:
```

Comment:
Bit of a nitpick, but since `key` is Hashable, it returns a Hashable as well (str.lower always returns a Hashable str)

```suggestion
    def _k(cls, key: Hashable) -> Hashable:
```